src/serializer/types.js

```diff
@@ -9,22 +9,31 @@
 
 /* @flow */
 
-import { DeclarativeEnvironmentRecord } from "../environment.js";
+import { DeclarativeEnvironmentRecord, type Binding } from "../environment.js";
 import { ConcreteValue, Value } from "../values/index.js";
-import type { ECMAScriptSourceFunctionValue } from "../values/index.js";
+import type { ECMAScriptSourceFunctionValue, FunctionValue } from "../values/index.js";
 import type { BabelNodeExpression, BabelNodeStatement } from "babel-types";
 import { SameValue } from "../methods/abstract.js";
 import { Realm } from "../realm.js";
 import invariant from "../invariant.js";
 
 export type TryQuery<T> = (f: () => T, defaultValue: T, logFailures: boolean) => T;
 
+export type AdditionalFunctionInfo = {
+  functionValue: FunctionValue,
+  captures: Names,
+  // TODO: use for storing modified residual function bindings (captured by other functions)
+  modifiedBindings: Map<Binding, ResidualFunctionBinding>,
+  instance: FunctionInstance,
+};
+
 export type FunctionInstance = {
   residualFunctionBindings: Map<string, ResidualFunctionBinding>,
   functionValue: ECMAScriptSourceFunctionValue,
   insertionPoint?: BodyReference,
   // Optional place to put the function declaration
   preludeOverride?: Array<BabelNodeStatement>,
+  additionalFunction?: FunctionValue,
```

Comment:
What exactly is the meaning of this? Couldn't the same function instance be referenced by multiple additional functions, and then what?

Reply:
Probably should rename this, it is the additional function that created the FunctionInstance (for when a function declaration was nested in an additional function)